### PR TITLE
Gaiaflex governance documentation

### LINKED ISSUE #54: Document how to use governance votes in gaia flex
Rough info is here: https://github.com/CosmWasm/wasmd/blob/master/x/wasm/Governance.md

Ask @alpe if you need more details.

- [x] Start up simple devnet with gaiaflex binary (and fast, eg. 10 minute governance cycle)
- [x] Try uploading contracts (via governance)
- [x] Try instantiateing contracts (via governance)
- [ ] Try executing contracts normally
- [ ] Document all steps.

(you can use the cw20 contract for examples)

### PR DESCRIPTION
**Pull request by orkunkl**
Resolves https://github.com/CosmWasm/testnets/issues/54
I've written an introduction and gave out cli samples. I need to fill in more details but I am just new to wasm and cosmos governance. Can you give me some tips on what to add?

### CODE REVIEW

File: gaiaflex/giving-life-smart-contract.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+`--run-as` flag is the contract creator, and it is required.
+
+If you run `gaiaflex tx gov submit-proposal wasm-store -h`, you will notice two more important flags:
+
+```shell
+--instantiate-everybody string      Everybody can instantiate a contract from the code, optional
+--instantiate-only-address string   Only this address can instantiate a contract instance from the code, optional
+```
+
+By default the first flag is enabled. If you want only one address to be able to initiate the contract,
```
Comment (ethanfrey) [inline gaiaflex/giving-life-smart-contract.md]: By default no one can instantiate this proposal except a governance vote.

If either of these flags are set, the voting committee should decide if that is acceptable for the given contract. Instantiate-everybody might make sense for a multisig (everyone makes their own), but not for creating a new token.

^^^^ Good to add some of this text

File: gaiaflex/giving-life-smart-contract.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+### Instantiating Contract
+
+Now the wasm code is stored to the chain after the voting period is ended and result is yes. A one more voting needs to be done
+to initiate the contract from the code with arguments.
+
+```shell script
+INIT='{"name":"Golden Stars","symbol":"STAR","decimals":2,"initial_balances":[{"address":"cosmos1x200a23zc2acc22mmy9pu4fuacl79w0yj4leqr","amount":"10000"},{"address":"cosmos1lj0cuh34c5useycd2wl3puqfr57lxd39hn8qcv","amount":"10000"}],"mint":{"minter":"cosmos1lj0cuh34c5useycd2wl3puqfr57lxd39hn8qcv"}}'
+gaiaflex tx gov submit-proposal instantiate-contract $PROPOSAL_ID $INIT --label "Init cw20 token" \
```
Comment (ethanfrey) [inline gaiaflex/giving-life-smart-contract.md]: "$INIT"

Comment (ethanfrey) [reply gaiaflex/giving-life-smart-contract.md]: label can be more telling, like "STAR" or "Golden Stars"

File: gaiaflex/giving-life-smart-contract.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    --run-as cosmos1lj0cuh34c5useycd2wl3puqfr57lxd39hn8qcv \
+    --title instantiate --description test \
+    --amount 100000umuon --deposit 101111umuon --from acc --gas auto --gas-adjustment=1.12 --gas-prices
+```
+
+Vote: ```gaiaflex tx gov vote 2 yes --from acc --gas auto --gas-prices "0.025umuon"```
+
+## Execute the contract
+
+Execution of a contract same as in [CW20 tutorial](../learn/using-contracts) section. Main difference is: TODO
```
Comment (ethanfrey) [inline gaiaflex/giving-life-smart-contract.md]: There should be no difference once it is instantiated, just a different codeId and contractAddr

File: gaiaflex/giving-life-smart-contract.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+## Migrating Contract
+
+**Documentation WIP**
+
+Here is a code migration contract if contracts supports the feature.
```
Comment (ethanfrey) [inline gaiaflex/giving-life-smart-contract.md:93]: Replace WIP with something like

Note: this is an advanced feature, both on the contract design as well as governance, and will be demoed by the CosmWasm team

File: gaiaflex/giving-life-smart-contract.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+```shell script
+gaiaflex tx gov submit-proposal migrate-contract $CONTRACT_ADDRESS $NEW_CODE_ID $MIGRATION_ARGS --title burner \
+  --description "perform migration" --run-as $(gaiaflex keys show -a code-runner) \ --deposit 1000umuon  --from acc \
+  --gas auto --gas-prices="0.025umuon"
+````
+
+## Set Admin
+
+A live contract's admin can be changed later with another voting:
```
Comment (ethanfrey) [inline gaiaflex/giving-life-smart-contract.md:106]: Explain that an "admin" is the one that can change the contract.

If it was set during instantiation, you don't even need a governance vote to modify the contract (like ethereum proxy libraries)

**Review by ethanfrey — APPROVED** [review]
Nice start. A little cleanup and you can merge.

We will get feedback when people try these out.

**Comment by orkunkl** [comment]
Merging if no other recommendations @ethanfrey.

**Pull request merged**